packages/blocks/src/time-picker/index.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { RawHTML, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import classnames from 'classnames';
+import { isEmpty } from 'lodash';
+/**
+ * Internal dependencies
+ */
+import { useColorStyles } from '@crowdsignal/styles';
+import {
+	ErrorMessage,
+	FormInputWrapper,
+	FormDateTimePicker,
+} from '../components';
+import { useField } from '@crowdsignal/form';
+
+const TimePicker = ( { attributes, className } ) => {
+	const { error, onChange, fieldValue } = useField( {
+		fieldName: `q_${ attributes.clientId }[text]`,
+		// initialValue: new Date().toDateString(),
+		validation: ( value ) => {
+			if ( attributes.mandatory && isEmpty( value ) ) {
+				return __( 'This field is required', 'blocks' );
+			}
+		},
+	} );
+
+	const handleTimeChange = ( e ) => {
+		onChange( e.toTimeString() );
+	};
+
+	const parsedTime = useMemo( () => {
+		// the DatePicker component has to have a fully formatted date, adding an arbitrary date to value
+		return Date.parse( 'Thu Sep 29 2022 ' + fieldValue );
```

Comment:
Since we just care about the time part of the string, there's no harm in using a hard-coded date. It's no a good practice, though.
If we change this logic to:

```JavaScript
	const parsedTime = useMemo( () => {
		if ( ! isEmpty( fieldValue ) ) {
			return  new Date( `${ new Date().toDateString() } ${ fieldValue }` );
		}
	}, [ fieldValue ] );
```

We get rid of the hard-coded string, and we also fix an issue where the initial value of the field is always set, instead of showing the placeholder.